sdk/python/kfp/_credentials.py

```diff
@@ -42,3 +50,38 @@ def refresh_api_key_hook(self, config: Configuration):
     @abc.abstractmethod
     def get_token(self):
         raise NotImplementedError()
+
+
+def read_sa_token(path=None):
+    """Read a ServiceAccount token found under some path."""
+    token = None
+    with open(path, "r") as f:
+        token = f.read().strip()
+    return token
+
+
+class ServiceAccountTokenVolumeCredentials(TokenCredentialsBase):
```

Comment:
Can we move this class to a separate package? To make the separation clear between interface and implementation.

Also all implementations should be optional.

Reply:
I split everything to distinct modules under the `kfp.auth` module.

I don't understand what it is that you're referring to by this:
> Also all implementations should be optional